hazelcast-client/src/test/java/com/hazelcast/client/quorum/list/ClientListReadQuorumTest.java

```diff
@@ -0,0 +1,50 @@
+package com.hazelcast.client.quorum.list;
+
+import com.hazelcast.client.quorum.ClientQuorumHelper;
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.config.Config;
+import com.hazelcast.core.IList;
+import com.hazelcast.instance.HazelcastInstanceFactory;
+import com.hazelcast.quorum.PartitionedCluster;
+import com.hazelcast.quorum.list.ListReadQuorumTest;
+import com.hazelcast.test.HazelcastParametersRunnerFactory;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static com.hazelcast.quorum.QuorumType.READ;
+import static com.hazelcast.quorum.QuorumType.READ_WRITE;
+import static com.hazelcast.quorum.QuorumType.WRITE;
+
+@RunWith(Parameterized.class)
+@Parameterized.UseParametersRunnerFactory(HazelcastParametersRunnerFactory.class)
+@Category({QuickTest.class})
+public class ClientListReadQuorumTest extends ListReadQuorumTest {
+
+    private static ClientQuorumHelper clients;
+
+    @Before
+    public void initialize() {
+        synchronized (lock) {
+            if (cluster == null) {
+                TestHazelcastFactory factory = new TestHazelcastFactory();
+                initCluster(PartitionedCluster.createClusterConfig(new Config()), factory, READ, WRITE, READ_WRITE);
+                clients = new ClientQuorumHelper(cluster, factory);
```

Comment:
I don't fully understand which threads are interacting here, but if a synchronized block is used it means that multiple threads could be running this method. In that case there is a risk that a different thread is using the 'clients' fields than the one that initialized it and there is no happens before relation since the field isn't volatile and isn't always obtained using the lock on 'lock'.

Reply:
These were all left-overs in the tests. All of these stuff is removed, should be clean and slim now. Thx